stackit/internal/services/dns/recordset/resource.go

```diff
@@ -264,8 +278,19 @@ func (r *recordSetResource) Read(ctx context.Context, req resource.ReadRequest,
 	ctx = tflog.SetField(ctx, "zone_id", zoneId)
 	ctx = tflog.SetField(ctx, "record_set_id", recordSetId)
 
+	if recordSetId == "" || zoneId == "" || projectId == "" {
+		tflog.Info(ctx, "Record set ID, zone ID, or project ID is empty, removing resource")
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
 	recordSetResp, err := r.client.GetRecordSet(ctx, projectId, zoneId, recordSetId).Execute()
 	if err != nil {
+		oapiErr, ok := err.(*oapierror.GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
```

Comment:
```suggestion
		var oapiErr *oapierror.GenericOpenAPIError
	    ok := errors.As(err, &oapiErr)
```

saves you the nolint comment ^^

Reply:
Thanks for the hint. Used the errors.As now